homeassistant/components/gpslogger/device_tracker.py

```diff
@@ -48,7 +48,7 @@ def _receive_data(device, gps, battery, accuracy, attrs):
     dev_reg = dr.async_get(hass)
     dev_ids = {
         identifier[1]
-        for device in dev_reg.devices.values()
+        for device in dev_reg.devices.get_devices_for_config_entry_id(entry.entry_id)
         for identifier in device.identifiers
         if identifier[0] == GPL_DOMAIN
```

Comment:
```suggestion
```
Same as for `geofancy`: https://github.com/home-assistant/core/pull/114810#discussion_r1551349258
The identifier should always match as we filter for the `entry.entry_id`.

Reply:
27e18108082